prov/rxm/src/rxm_ep.c

```diff
@@ -1001,11 +1019,15 @@ rxm_ep_sar_tx_prepare_and_send_segment(struct rxm_ep *rxm_ep, struct rxm_conn *r
 			/* if the sending for the first segment fails,
 			 * release resources and report this to user */
 			rxm_tx_buf_release(tx_entry->ep, tx_buf);
+			if (!dlist_empty(&tx_entry->deferred_tx_entry))
+				rxm_ep_dequeue_deferred_tx_queue(tx_entry);
 			rxm_tx_entry_release(rxm_conn->send_queue, tx_entry);
 			return -FI_EAGAIN;
 		}
 		dlist_insert_tail(&tx_buf->hdr.entry,
-				  &tx_entry->deferred_tx_buf_list);	
+				  &tx_entry->deferred_tx_buf_list);
+		if (dlist_empty(&tx_entry->deferred_tx_entry))
```

Comment:
Shouldn't this be always true? Once we add to deferred_tx_buf_list, we would always return in line 1011.